src/test/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RouteRequestTransitDataProviderFilterTest.java

```diff
@@ -264,8 +264,121 @@ public void bannedTripFilteringTest() {
     assertFalse(valid);
   }
 
+  /**
+   * Filtering trips with 2 filters: a selection by submode filter and an exclusion by agency filter.
+   * A trip matches if it matches either of them.
```

Comment:
Do you mean that the trip either matches submode filter or does not match excluded agency?

Reply:
Yes, a trip matches if its submode is in the list of selected submodes OR if its agency is NOT in the list of excluded agencies.